src/clj/qbits/jet/servlet.clj

```diff
@@ -50,21 +49,26 @@
 (defn build-request-map
   "Create the request map from the HttpServletRequest object."
   [^Request request]
-  {:servlet-request    request
-   :server-port        (.getServerPort request)
-   :server-name        (.getServerName request)
-   :remote-addr        (.getRemoteAddr request)
-   :uri                (.getRequestURI request)
-   :query-string       (.getQueryString request)
-   :scheme             (keyword (.getScheme request))
-   :request-method     (keyword (.toLowerCase (.getMethod request)))
-   :headers            (get-headers request)
-   :content-type       (.getContentType request)
-   :content-length     (get-content-length request)
-   :character-encoding (.getCharacterEncoding request)
-   :ssl-client-cert    (get-client-cert request)
-   :ctrl               (async/chan)
-   :body               (.getInputStream request)})
+  (let [request-headers (get-headers request)]
+    {:servlet-request request
+     :server-port (.getServerPort request)
+     :server-name (.getServerName request)
+     :remote-addr (.getRemoteAddr request)
+     :uri (.getRequestURI request)
+     :query-string (.getQueryString request)
+     :scheme (keyword (.getScheme request))
+     :request-method (keyword (.toLowerCase (.getMethod request)))
+     :headers request-headers
+     :trailers-fn (when (util/trailers-supported? (.getProtocol request) request-headers)
```

Comment:
Should we handled trailers via a `chan` for consistency?  Should users of the client providing trailers be able to block while trailers are being generated?

Reply:
Also, I did not use trailers since there is no event notification mechanism on the org.eclipse.jetty.server.Request object for when trailers become available.